scripts/release/build/update-package-versions.js

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env node
+
+'use strict';
+
+const chalk = require('chalk');
+const {exec} = require('child-process-promise');
+const {readFileSync, writeFileSync} = require('fs');
+const {readJson, writeJson} = require('fs-extra');
+const {join} = require('path');
+const {projects} = require('../config');
+const {execUnlessDry, logPromise} = require('../utils');
+
+const update = async ({cwd, dry, version}) => {
+  try {
+    // Update root package.json
+    const packagePath = join(cwd, 'package.json');
+    const rootPackage = await readJson(packagePath);
+    rootPackage.version = version;
+    await writeJson(packagePath, rootPackage, {spaces: 2});
+
+    // Update ReactVersion source file
+    const reactVersionPath = join(cwd, 'src/ReactVersion.js');
+    const reactVersion = readFileSync(reactVersionPath, 'utf8').replace(
+      /module\.exports = '[^']+';/,
+      `module.exports = '${version}';`
+    );
+    writeFileSync(reactVersionPath, reactVersion);
+
+    // Update renderer versions and peer dependencies
+    const updateProjectPackage = async project => {
+      const path = join(cwd, 'packages', project, 'package.json');
+      const json = await readJson(path);
+      json.version = version;
+      if (project !== 'react') {
+        json.peerDependencies.react = `^${version}`;
+      }
+      await writeJson(path, json, {spaces: 2});
+    };
+    await Promise.all(projects.map(updateProjectPackage));
+
+    // Version sanity check
+    await exec('yarn version-check', {cwd});
```

Comment:
Gotcha - at first was confused because this is similar to `yarn check` but I see it's our own script here (https://github.com/facebook/react/blob/master/scripts/tasks/version-check.js) which will throw if the versions don't match. Nice.